File: frontend/src/types/executiveDashboard.ts
export interface DomainReadiness {
  data_source_id: number;
  domain_name: string;
  database_type: string;

  readiness_score: number;
  maturity_level: string;
  status: string;

  approved_knowledge_percentage: number;
  approved_dictionary_percentage: number;
  governance_percentage: number;

  requires_attention: boolean;
  attention_reason: string | null;
}

export interface ExecutiveHighlight {
  severity:
    | "success"
    | "warning"
    | "info"
    | "error";

  title: string;
  message: string;
}

export interface ExecutiveDashboard {
  enterprise_readiness: number;
  readiness_label: string;

  connected_business_systems: number;
  ai_ready_business_systems: number;
  systems_requiring_attention: number;

  governance_compliance: number;
  knowledge_maturity: number;

  domains: DomainReadiness[];
  highlights: ExecutiveHighlight[];
}

File: backend/app/schemas/executive_dashboard.py
from pydantic import BaseModel


class DomainReadinessResponse(BaseModel):
    data_source_id: int
    domain_name: str
    database_type: str

    readiness_score: int
    maturity_level: str
    status: str

    approved_knowledge_percentage: int
    approved_dictionary_percentage: int
    governance_percentage: int

    requires_attention: bool
    attention_reason: str | None


class ExecutiveHighlightResponse(BaseModel):
    severity: str
    title: str
    message: str


class ExecutiveDashboardResponse(BaseModel):
    enterprise_readiness: int
    readiness_label: str

    connected_business_systems: int
    ai_ready_business_systems: int
    systems_requiring_attention: int

    governance_compliance: int
    knowledge_maturity: int

    domains: list[DomainReadinessResponse]
    highlights: list[ExecutiveHighlightResponse]
